Adding three pieces of code to the github to run my jet vn analysis u…

wwitt4 opened the pull request:
sing the train system. - Will Witt

alibuild commented:
b4ccc363562177f54fc1e00d3c1488e9d79e06d6: approval required: 1 of @davidrohr (David Rohr), @loizides (Constantinos Loizides), @dcaffarr (Davide Caffarri), @jgrosseo (Jan Fiete Grosse-Oetringhaus), @raymondEhlers (Raymond James Ehlers), @qgp (Jochen Klein), @adriansev (Adrian Sevcenco), @rhaake (Rudiger Haake), @mfasDa (Markus Fasel), @matplo (Mateusz Ploskon), @lcunquei (Leticia Cunqueiro Mendez), @vkucera (Vit Kucera), @shahor02 (Ruben Shahoyan)

_Comment with `+1` to approve and allow automatic merging,or with `+test` to run tests only. **Please comment on the pull request: [click here](https://github.com/alisw/AliPhysics/pull/14917) and comment at the bottom of the page.**_

raymondEhlers commented on PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskJetVn.h:
```diff
@@ -0,0 +1,96 @@
+  void                        UserCreateOutputObjects()                         ;
+  void                        Terminate(Option_t *option)                       ;
+
+  static AliAnalysisTaskJetVn* AddTaskEmcalJetSample(
```
Although this will run fine, it's super misleading. I suggest something like `AddTaskJetVn` or whatever you like. Just don't forget to update the AddTask

raymondEhlers commented on PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskJetVn.h:
```diff
@@ -0,0 +1,96 @@
+  Double_t param1;
+
+ protected:
+  AliEventCuts fEventCuts                                        ;
```
You should specify what is and isn't streaming

alibuild commented:
4e619c8d2b57d75a599cbf3b9436a6e3ade5d004: approval required: 1 of @davidrohr (David Rohr), @loizides (Constantinos Loizides), @dcaffarr (Davide Caffarri), @jgrosseo (Jan Fiete Grosse-Oetringhaus), @raymondEhlers (Raymond James Ehlers), @qgp (Jochen Klein), @adriansev (Adrian Sevcenco), @rhaake (Rudiger Haake), @mfasDa (Markus Fasel), @matplo (Mateusz Ploskon), @lcunquei (Leticia Cunqueiro Mendez), @vkucera (Vit Kucera), @shahor02 (Ruben Shahoyan)

_Comment with `+1` to approve and allow automatic merging,or with `+test` to run tests only. **Please comment on the pull request: [click here](https://github.com/alisw/AliPhysics/pull/14917) and comment at the bottom of the page.**_

wwitt4 commented:
Closing this pull request to resubmit with Raymond's corrections.

raymondEhlers commented:
@wwitt4 since you pushed to the same branch, it will update the pull request. You can just reopen this one

wwitt4 commented:
@raymondEhlers Ah, I didn't know that. First time submitting to the AliPhysics Github. I have updated the CMakeList and LinksDef files to include my jet vn code, changed the AddTask function name, and added the streamer comments. Does it look ready for inclusion in AliPhysics?

raymondEhlers commented:
@wwitt4 Yeah, I guessed that might be the case - no problem. The changes look good, except that you still have one small issue: you applied the streamer comments to your class methods instead of the members. We could just commit it now, but I think you'll be much better off if you take 5 minutes to add the streamer comments to the four members that are missing them (`param0`, `param1`, `fEventCuts`, and `fFlowQnVectorMgr`. I assume that you want to stream the first 3 but not the fourth)

A few other side notes (which you don't need to change right now - just for your knowledge:
- AliEventCuts was integrated into the EMCal jet framework - you don't need to define it in your class anymore. But it's also fine as is
- You should use getters and setters for `param0` and `param1`, but for now it doesn't matter

alibuild commented:
c7e2d1da39568850e655fff02b373682f1023f2d: approval required: 1 of @davidrohr (David Rohr), @loizides (Constantinos Loizides), @dcaffarr (Davide Caffarri), @jgrosseo (Jan Fiete Grosse-Oetringhaus), @raymondEhlers (Raymond James Ehlers), @qgp (Jochen Klein), @adriansev (Adrian Sevcenco), @rhaake (Rudiger Haake), @mfasDa (Markus Fasel), @matplo (Mateusz Ploskon), @lcunquei (Leticia Cunqueiro Mendez), @vkucera (Vit Kucera), @shahor02 (Ruben Shahoyan)

_Comment with `+1` to approve and allow automatic merging,or with `+test` to run tests only. **Please comment on the pull request: [click here](https://github.com/alisw/AliPhysics/pull/14917) and comment at the bottom of the page.**_

raymondEhlers commented:
+1

alibuild commented:
c7e2d1da39568850e655fff02b373682f1023f2d: approved: will be automatically merged on successful tests

mfasDa commented:
There are couple of compiler warnings in the new task. Please have a look and fix, no matter whether they are critical or not!

DEBUG:AliPhysics:AliPhysics:aliceo2: /Users/markusfasel/alice/sw/SOURCES/AliPhysics/aliceo2/0/PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskJetVn.cxx:511:14: warning: unused variable 'psi3' [-Wunused-variable]
DEBUG:AliPhysics:AliPhysics:aliceo2:     Double_t psi3, phiMinusPsi3;
DEBUG:AliPhysics:AliPhysics:aliceo2:              ^
DEBUG:AliPhysics:AliPhysics:aliceo2: /Users/markusfasel/alice/sw/SOURCES/AliPhysics/aliceo2/0/PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskJetVn.cxx:510:14: warning: unused variable 'psi2' [-Wunused-variable]
DEBUG:AliPhysics:AliPhysics:aliceo2:     Double_t psi2, phiMinusPsi2; // angles for jet vn calculation
DEBUG:AliPhysics:AliPhysics:aliceo2:              ^
DEBUG:AliPhysics:AliPhysics:aliceo2: /Users/markusfasel/alice/sw/SOURCES/AliPhysics/aliceo2/0/PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskJetVn.cxx:663:14: warning: unused variable 'psi3' [-Wunused-variable]
DEBUG:AliPhysics:AliPhysics:aliceo2:     Double_t psi3, phiMinusPsi3;
DEBUG:AliPhysics:AliPhysics:aliceo2:              ^
DEBUG:AliPhysics:AliPhysics:aliceo2: /Users/markusfasel/alice/sw/SOURCES/AliPhysics/aliceo2/0/PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskJetVn.cxx:662:14: warning: unused variable 'psi2' [-Wunused-variable]
DEBUG:AliPhysics:AliPhysics:aliceo2:     Double_t psi2, phiMinusPsi2; // angles for jet vn calculation
DEBUG:AliPhysics:AliPhysics:aliceo2:              ^
DEBUG:AliPhysics:AliPhysics:aliceo2: /Users/markusfasel/alice/sw/SOURCES/AliPhysics/aliceo2/0/PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskJetVn.cxx:886:10: warning: unused variable 'sumAcceptedTracks' [-Wunused-variable]
DEBUG:AliPhysics:AliPhysics:aliceo2:   UInt_t sumAcceptedTracks = 0;
DEBUG:AliPhysics:AliPhysics:aliceo2:          ^
DEBUG:AliPhysics:AliPhysics:aliceo2: 5 warnings generated.
